### DISCUSSION TITLE
How Renovate detects the project type with `rangeStrategy: auto`?

### DISCUSSION THREAD

**Discussion by chalkygames123** [renovatebot/renovate#9446] [Request Help] [answered]
Hi,

In my repository https://github.com/chalkygames123/eslint-config, even though it's a library, Renovate creates a [Pin PR](https://github.com/chalkygames123/eslint-config/pull/1) for dependencies.

I guess the reason is that the `package.json#main` field is omitted. But if so, I think the current behavior is incorrect because the [npm Docs](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#main) says:

> If `main` is not set it defaults to `index.js` in the packages root folder.

Anyway, it would be nice to have more detailed documentation on the detection with the "auto" strategy.

**Comment by rarkins** [comment, marked as answer]
The logic is here: https://github.com/renovatebot/renovate/blob/master/lib/manager/npm/extract/type.ts

**Reply by lorentzlasson** [reply]
I'm guessing this is the new location https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/npm/extract/type.ts

**Comment by viceice** [comment]
Configure [rangeStrategy](https://docs.renovatebot.com/configuration-options/#rangestrategy) to a supported value, that should disable auto pin pr

**Comment by chalkygames123** [comment]
Thank you all for the quick responses.

I just reached https://github.com/renovatebot/renovate/issues/4640 and decided to add the `main` field for now because it seems there are no easy solutions.
